torch/_inductor/select_algorithm.py

```diff
@@ -2353,26 +2353,54 @@ def __init__(
         self.kwargs = kwargs or {}
         self.has_out_variant = has_out_variant
         self.gm = choice.gm
+        self.bmreq: Optional[BenchmarkRequest] = None
+
+        from torch._inductor.autotune_process import (
+            ExternKernelBenchmarkRequest,
+            ExternKernelCPUBenchmarkRequest,
+            ExternKernelGPUBenchmarkRequest,
+        )
+
+        # Determine if this is a GPU or CPU kernel
+        if self.layout:
+            device = self.layout.device
+        else:
+            device = None
+            for inp_node in self.input_nodes:
+                dev = inp_node.get_device()
+                if dev and dev.type != "cpu":
+                    device = dev
+                    break
+
+            if not device:
+                device = torch.device("cpu")
+
+        self.input_tensor_meta: Union[list[TensorMeta], TensorMeta]
+        self.output_tensor_meta: Union[list[TensorMeta], TensorMeta]
+        if device.type == "cpu":
+            self.input_tensor_meta, self.output_tensor_meta = [], []
+            benchmark_cls = ExternKernelCPUBenchmarkRequest
+        else:
+            self.input_tensor_meta = TensorMeta.from_irnodes(self.input_nodes)
```

Comment:
Hi @PaulZhang12,

This PR is causing a large number of failures on XPU. For example: https://hud.pytorch.org/pr/pytorch/pytorch/168328#57614268055
. In that case, self.input_nodes contains a TensorBox(StorageBox(Pointwise)), which does not implement get_stride, leading to runtime errors.

In addition, the hard-coded CUDA logic in _do_bench_using_profiling here:
https://github.com/pytorch/pytorch/blob/afb173d9b9440d804b5f77d0c291e53c720d1fcf/torch/_inductor/utils.py#L318

also causes many failures on XPU.

Given the impact, could we revert this PR for the upcoming release?
If needed, I’m happy to help add proper support for _do_bench_using_profiling on XPU.

FYI @EikanWang